pkgs/by-name/mu/multipass/update.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -I nixpkgs=./. -i bash -p curl jq nix-prefetch-github yj git
+# shellcheck shell=bash
+
+set -euo pipefail
+
+multipass_dir="./pkgs/by-name/mu/multipass"
+
+# Download the latest pubspec.lock (which is a YAML file), convert it to JSON and write it to
+# the package directory as pubspec.lock.json
+update_pubspec_json() {
+    local version; version="$1"
+    echo "Updating pubspec.lock.json"
+
+    curl -s \
+        "https://raw.githubusercontent.com/canonical/multipass/refs/tags/v${version}/src/client/gui/pubspec.lock" \
+        | yj > "$multipass_dir/pubspec.lock.json"
+}
+
+# Update the SRI hash of a particular overridden Dart package in the Nix expression
+update_dart_pkg_hash() {
+    local pkg_name; pkg_name="$1"
+    local owner; owner="$2";
+    local repo; repo="$3";
+    echo "Updating dart package hash: $pkg_name"
+
+    resolved_ref="$(jq -r --arg PKG "$pkg_name" '.packages[$PKG].description."resolved-ref"' "$multipass_dir/pubspec.lock.json")"
+    hash="$(nix-prefetch-github --rev "$resolved_ref" "$owner" "$repo" | jq '.hash')"
+    sed -i "s|${pkg_name} = \".*\";|${pkg_name} = $hash;|" "$multipass_dir/gui.nix"
+}
+
+# Update the hash of the multipass source code in the Nix expression.
+# This is not an ideal method of obtaining the new hash, but nix-prefetch doesn't work
+# with postFetch: https://github.com/NixOS/nix/issues/1514
+update_multipass_source() {
+    local version; version="$1"
+    echo "Updating multipass source"
+
+    dir="$(mktemp -d)";
+    git clone -b v"$version" --recurse-submodules https://github.com/canonical/multipass "$dir"
+
+    rm -rf "$dir/.git"
+    base_hash="$(nix hash path "$dir")"
+    base32_hash="$(nix hash convert --to nix32 "$base_hash")"
+    sri_hash="$(nix hash convert --to sri "sha256:$base32_hash")"
+
+    sed -i "s|version = \".*$|version = \"$version\";|" "$multipass_dir/package.nix"
+    sed -i "s|hash = \".*$|hash = \"${sri_hash}\";|" "$multipass_dir/package.nix"
+}
+
+# Update the version/hash of the grpc source code in the Nix expression.
+update_grpc_source() {
+    local version; version="$1"
+    echo "Updating grpc source"
+
+    submodule_info="https://raw.githubusercontent.com/canonical/multipass/refs/tags/v${version}/3rd-party/submodule_info.md"
+    commit_short="$(curl -s "${submodule_info}" | grep -Po "CanonicalLtd/grpc/compare/v[0-9]+\.[0-9]+\.[0-9]+\.\.\K[0-9a-f]+")"
+    commit_hash="$(curl -s "https://github.com/canonical/grpc/commits/${commit_short}" | grep -Po "${commit_short}[0-9a-f]+" | head -n1)"
+    sri_hash="$(nix-prefetch-github canonical grpc --rev "${commit_hash}" --fetch-submodules | jq -r '.hash')"
+
+    sed -i "s|rev = \".*$|rev = \"${commit_hash}\";|" "$multipass_dir/multipassd.nix"
+    sed -i "s|hash = \".*$|hash = \"${sri_hash}\";|" "$multipass_dir/multipassd.nix"
+}
+
+
+LATEST_TAG="$(curl -s https://api.github.com/repos/canonical/multipass/releases/latest | jq -r '.tag_name')"
```

Comment:
This should optionally take a GITHUB_TOKEN. Please see other update scripts on how to do that.